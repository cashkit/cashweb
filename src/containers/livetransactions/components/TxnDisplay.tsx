import React, { Profiler } from 'react';

export interface TxnDisplayInterface {
  transaction: string,
  onClickTx: Function
}

/**
 * Use the profiler to check for rerenders.
 * Usage:
 *  <Profiler id={transaction} onRender={onRenderProfilerCallback}>
 *    <p key={transaction}>
 *       transaction: <a onClick={() => onClickTx(transaction)} className="content">{transaction}</a>
 *    </p>
 *  </Profiler>
 * 
 * @param id - the "id" prop of the Profiler tree that has just committed.
 * @param phase - either "mount" (if the tree just mounted) or "update" (if it re-rendered).
 * @param actualDuration - time spent rendering the committed update
 * @param baseDuration - estimated time to render the entire subtree without memoization
 * @param startTime - when React began rendering this update
 * @param commitTime - when React committed this update
 * @param interactions - the Set of interactions belonging to this update
 */
  const onRenderProfilerCallback = (id, phase, actualDuration, baseDuration, startTime, commitTime, interactions) => {
  console.log(`[INFO] PROFILER: ID: ${id}
    PHASE: ${phase}
    ACTUAL_DURATION: ${actualDuration}, 
    BASE_DURATION: ${baseDuration}
    START_TIME: ${startTime}
    COMMIT_TIME: ${commitTime}
    INTERACTIONS: ${interactions.size}`
  )
}

/**
 * @param transaction: Transaction hash
 * @param onClickTx: Callback function for click event on transaction hash.
 * @returns 
*/
export const TxnDisplay = ({transaction, onClickTx}: TxnDisplayInterface ) => {
  return (
      <p key={transaction}>
        transaction: <a onClick={() => onClickTx(transaction)} className="content">{transaction}</a>
      </p>
  )
}

function areEqual(prevProps, nextProps) {
  /*
  return true if passing nextProps to render would return
  the same result as passing prevProps to render,
  otherwise return false
  */
 if (prevProps.transaction == nextProps.transaction){
   return true
 }
 return false
}

export const MemoizedTxnDisplay = React.memo(TxnDisplay, areEqual);
