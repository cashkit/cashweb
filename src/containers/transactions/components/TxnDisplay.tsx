import React from 'react';

export interface TxnDisplayInterface {
  transaction: string,
  onClickTransaction: Function
}

/**
 * @param transaction: Transaction hash
 * @param onClickTransaction: Callback function for click event on transaction hash.
 * @returns 
*/
export const TxnDisplay = ({transaction, onClickTransaction}: TxnDisplayInterface ) => {
  return (
    <p key={transaction}>
      transaction: <a onClick={() => onClickTransaction(transaction)} className="content">{transaction}</a>
    </p>
  )
}