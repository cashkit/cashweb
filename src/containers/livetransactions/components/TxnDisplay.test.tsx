import React from 'react';
import renderer from 'react-test-renderer';
import { TxnDisplay, TxnDisplayInterface } from './TxnDisplay';

const props: TxnDisplayInterface = {
    transaction: "",
    onClickTx: () => {}
}

it('<TxnDisplay/> renders correctly', () => {
    const tree = renderer
        .create(<TxnDisplay {...props}></TxnDisplay>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});