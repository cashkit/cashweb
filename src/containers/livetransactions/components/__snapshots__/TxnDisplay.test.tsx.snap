// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TxnDisplay/> renders correctly 1`] = `
<p>
  transaction: 
  <a
    className="content"
    onClick={[Function]}
  >
    
  </a>
</p>
`;
